sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -335,14 +338,19 @@ export class NiseRecorder extends BaseRecorder {
     this.environmentSetup = recorderEnvironmentSetup;
     const self = this;
     const xhr = nise.fakeXhr.useFakeXMLHttpRequest();
+    this.xhr = xhr;
 
     // 'karma-json-preprocessor' helps us to retrieve recordings
-    this.recordings = (window as any).__json__[
-      "recordings/" + this.relativeTestRecordingFilePath
-    ].recordings;
-    this.uniqueTestInfo = (window as any).__json__[
-      "recordings/" + this.relativeTestRecordingFilePath
-    ].uniqueTestInfo;
+    this.recordings = windowLens.get([
```

Comment:
I thought about this some more, and another way to accomplish this (without having to have fancy setter/getters) is something like

```ts
declare var __json__: RecordingMap;
```

or

```ts
interface Window {
    __json__: RecordingMap;
}
```

Then you don't have to worry about casting to `any`

Reply:
I think I prefer the idea of a central place to edit the window object though. Since I'd like to have a replacement of the window object in the future, somehow. At least a single place to change the window, so that we can track if something goes wrong. I can still use your ideas to replace the "any" type in that central place, if that makes sense.

Reply:
I made this issue to track not using any: https://github.com/Azure/azure-sdk-for-js/issues/7402